shared/js/background/classes/tab.es6.js

```diff
@@ -81,8 +81,11 @@ class Tab {
         }
     }
 
-    updateSite () {
-        this.site = new Site(this.url)
+    updateSite (url) {
+        if (this.site.url === url) return
```

Comment:
`webNavigation.onCommitted` apparently happens after the server has started sending us data (https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webNavigation/onCommitted) so `onHeadersReceived` will have already happened.

For the vast majority of sites, this guard will return for the `onCommitted` call because the URL would be exactly the same.